### DISCUSSION TITLE
Get or Default?

### DISCUSSION THREAD

**Discussion by clansdown** [nlohmann/json#3917] [Q&A]
When dealing with json fields that are optional, it's a PITN to have to check a field exists before every access in order to default to a default value if it's not present. Is there any way to do a a get_or_default, something like:

myjson.["user"].get("canBuyTarantulas", false)

(where the type would come from the default that's provided, and if the key doesn't exist or is null, the default value is returned instead, and no exceptions are thrown?)

**Comment by nlohmann** [comment]
Yes, it's called value (https://json.nlohmann.me/api/basic_json/value/)

**Reply by clansdown** [reply]
Thanks! (I promise I tried to find it in the documentation, and then searched on the forum first.)

**Reply by clansdown** [reply]
By the way, this is just an excellent library and such a pleasure to use. Thank you, in general.
